Create working directory if it wasn't found

hatchan opened the pull request:
Look for .fpx directory in the current working directory ancestors, if it wasn't found, then create a .fpx directory in the current directory

Store libsql data in the fpx directory again.

Create a .fpx directory in the current repository as a placeholder, making it easier to develop for fpx

Resolves: FP-4017, FP-4018

hatchan commented on fiberplane-models/Cargo.toml:
```diff
@@ -13,10 +13,12 @@ version = "1.0.0-beta.10"
 fp-bindgen = ["base64uuid/fp-bindgen", "dep:fp-bindgen"]
 
 [dependencies]
+axum = { version = "0.6.1", optional = true }
```
This should implicitly add the `axum = ["dep:axum"]` feature, right?

hatchan commented on fiberplane-openapi-rust-gen/src/generator.rs:
```diff
@@ -185,9 +186,12 @@ fn add_dependencies(dependencies: &mut DepsSet, args: &GeneratorArgs) -> Result<
-
     dependencies.insert("bytes".to_string(), Dependency::Simple("1".to_string()));
 
+    dependencies.insert("thiserror".to_string(), Dependency::Simple("1".to_string()));
```
This is needed for the `ClientError`

hatchan commented on fiberplane-openapi-rust-gen/src/generator.rs:
```diff
@@ -185,9 +186,12 @@ fn add_dependencies(dependencies: &mut DepsSet, args: &GeneratorArgs) -> Result<
 
+    dependencies.insert("thiserror".to_string(), Dependency::Simple("1".to_string()));
+
+    dependencies.insert("url".to_string(), Dependency::Simple("2".to_string()));
```
This is needed to expose the `UrlParse` error

hatchan commented on fiberplane-openapi-rust-gen/src/routes.rs:
```diff
@@ -113,30 +116,43 @@ pub(crate) fn generate_routes(
 ) -> Result<()> {
+    let use_new_style = operation
+        .extensions
+        .get("x-new-style")
```
`x-new-style` should be used for any future endpoints. It will require a typed error and uses `PagedVec` for array endpoints.

mellowagain replied:
only in the openapi file i assume? or do we also need to include it on api responses?

hatchan replied:
The `x-new-style` is only for the openapi schema indeed. The API endpoints and client will use the `PagedVec` (for array's) and will return the typed errors.

hatchan commented on fiberplane/Cargo.toml:
```diff
@@ -14,6 +14,7 @@ crate-type = ["lib"]
 
 [features]
 api-client = ["fiberplane-api-client"]
+axum = ["fiberplane-models?/axum"]
```
I couldn't come up with a better name, other than `axum`. This is to add the IntoResponse to the typed error objects.

mellowagain commented on fiberplane-openapi-rust-gen/src/client_config.rs:
```diff
@@ -53,6 +53,24 @@ pub(crate) fn generate_client_configs(servers: &[Server], src_path: &Path) -> Re
+
+    writer
+        .flush()
+        .context("Failed to flush output for `builder.rs`")
```
```suggestion
        .context("Failed to flush output for `api_client.rs`")
```

hatchan commented on fiberplane-api-client/src/clients.rs:
```diff
@@ -20,43 +20,28 @@ pub fn default_config(
-        ApiClientBuilder::new(base_url)
-    }
-}
+#[deprecated(note = "Use `crate::ApiClient` instead")]
```
The only "problem" is that this depends if you are using it through the `fiberplane` "meta" crate or directly using the `fiberplane_api_client`. I guess we can put both suggestions there :thinking:

gagbo reviewed (approved):
The `axum` feature makes sense, but I’m afraid of potential compatibility issues downstream if they use a different version than `0.6`

hatchan commented on fiberplane-models/src/pagerduty.rs:
```diff
@@ -0,0 +1,357 @@
+    UpdatedAt,
+}
+
+impl PagerDutyReceiverListSortFields {
+    #[inline]
+    pub fn to_sql(&self) -> &'static str {
+        match self {
+            PagerDutyReceiverListSortFields::Name => "name",
+            PagerDutyReceiverListSortFields::CreatedAt => "created_at",
+            PagerDutyReceiverListSortFields::UpdatedAt => "updated_at",
+        }
+    }
+}
```
Yeah fair, tbh I don't think this implementation detail should be exposed in `fiberplane/fiberplane` at all. The reason why this is a separate function is that in some situations the sort field might not map directly to a sql column.

I'll remove this and just use the strum impl for now.

hatchan committed:
Create working directory if it wasn't found

Look for .fpx directory in the current working directory ancestors, if it wasn't found, then create a .fpx directory in the current directory

Store libsql data in the fpx directory again.

Create a .fpx directory in the current repository as a placeholder, making it easier to develop for fpx

hatchan committed:
Add debug command to print the location of the fpx-directory

hatchan committed:
Resolve clippies

hatchan committed:
Change comment for fpx_directory argument

hatchan committed:
Move fpx-directory sub command to the system command

hatchan committed:
Simplify find_fpx_dir by not returning a Result

Make internals of find_fpx_dir simpler by not iterating over _all_ entries and only check for the .fpx directory

'fpx system database delete' won't initialize the .fpx directory

Fix some comments

hatchan committed:
Resolve clippy
